contracts/governancestaking/BaseGovernanceShareField.sol

```diff
@@ -0,0 +1,203 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.6;
+import "../Interfaces.sol";
+import "openzeppelin-solidity/contracts/utils/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/utils/math/Math.sol";
+import "../utils/DAOContract.sol";
+import "../utils/DSMath.sol";
+import "hardhat/console.sol";
+
+contract BaseGovernanceShareField is DAOContract {
+	using SafeMath for uint256;
+
+	uint256 totalProductivity;
+	uint256 accAmountPerShare;
+
+	uint256 public mintedShare;
+	uint256 public mintCumulation;
+	address public shareToken;
+
+	uint256 public lastRewardBlock;
+    uint256 public rewardsPerBlock;
+
+	struct UserInfo {
+		uint256 amount; // How many tokens the user has provided.
+		uint256 rewardDebt; // Reward debt.
+		uint256 rewardEarn; // Reward earn and not minted
+	}
+
+	mapping(address => UserInfo) public users;
+	/**
+	 * @dev Function to set rewards for per block in GDAO so 0 decimals
+	 */
+	function setRewardsPerBlock (uint256 _rewardsPerBlock) public{
+        _onlyAvatar();
+        rewardsPerBlock = _rewardsPerBlock;
+    }
+
+	function _setShareToken(address _shareToken) internal {
+		shareToken = _shareToken;
+	}
+
+	/**
+	 * @dev Update reward variables of the given pool to be up-to-date.
+	 * Calculates passed blocks and adding to the reward pool
+	 */
+	function _update() internal virtual {
+		if (totalProductivity == 0) {
+			lastRewardBlock = block.number;
+			return;
+		}
+		
+		
+        uint256 multiplier = block.number - lastRewardBlock; // Blocks passed since last reward block
+        uint256 reward = multiplier * rewardsPerBlock; // rewardsPerBlock is in GDAO which is in 0 decimals
+
+        accAmountPerShare =accAmountPerShare + rdiv(reward * 1e2 ,totalProductivity) / 1e25; //multiply reward by 1e2 so bring it to 2 decimals as well then divide result by 1e25 so accAmountPerShare is in 2 decimals
```

Comment:
accAmountPerShare should have high precision, otherwise it could be 0.
we'll also change GDAO to have 18 decimals

Reply:
Yeah I was aware with it , was gonna ask